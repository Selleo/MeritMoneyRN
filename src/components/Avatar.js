import React, { Component } from 'react'
import { Platform, Image, View, Text, StyleSheet } from 'react-native'

import { white, primary } from 'src/styles/colors'

export default class Avatar extends Component {
  static defaultProps = {
    size: 50,
    source: 'https://i.imgur.com/ckqCXMS.png',
  }

  render() {
    const { size, source, showBadge, badgeCounter } = this.props
    const avatarStyles = {
      height: size,
      width: size,
      borderRadius: size / 2,
    }

    return (
      <View style={[styles.container, avatarStyles]}>
        {showBadge && (
          <View style={styles.badgeContainer}>
            <Text style={styles.badgeText}>{badgeCounter}</Text>
          </View>
        )}
        <Image source={{ uri: source }} style={[avatarStyles, styles.image]} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: white,
  },
  badgeContainer: {
    alignItems: 'center',
    backgroundColor: primary,
    borderRadius: 12.5,
    height: 25,
    overflow: 'hidden',
    justifyContent: 'center',
    position: 'absolute',
    width: 25,
    ...Platform.select({
      ios: {
        right: -5,
        top: 5,
      },
      android: {
        top: 0,
        left: 0,
      },
    }),
  },
  badgeText: {
    paddingTop: 2,
    fontFamily: 'Lato-Regular',
    fontSize: 16,
  },
  image: {
    zIndex: -1,
  },
})
