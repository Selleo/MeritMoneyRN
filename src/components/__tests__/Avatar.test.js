import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Avatar from '../Avatar'

describe('<Avatar />', () => {
  const defaultProps = {
    size: 50,
    source: 'https://i.imgur.com/ckqCXMS.png',
  }

  const wrapper = renderer.create(<Avatar {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
