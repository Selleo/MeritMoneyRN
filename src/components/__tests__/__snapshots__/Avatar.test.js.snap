// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
      },
      Object {
        "borderRadius": 25,
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.imgur.com/ckqCXMS.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 25,
          "height": 50,
          "width": 50,
        },
        Object {
          "zIndex": -1,
        },
      ]
    }
  />
</View>
`;
